File: 剑指offer/剑指 Offer 18. 删除链表的节点/deleteNode.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution:
    def deleteNode(self, head: ListNode, val: int) -> ListNode:
        dummyNode = ListNode(0, head)
        cur, pre = head, dummyNode
        while cur.val != val:
            pre, cur = pre.next, cur.next
        pre.next, cur = cur.next, None
        return dummyNode.next

File: 剑指offer/剑指 Offer 18. 删除链表的节点/deleteNode.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var deleteNode = function (head, val) {
  let dummy = new ListNode(0);
  dummy.next = head;
  let cur = head;
  let pre = dummy;
  while (cur.val !== val) {
    cur = cur.next;
    pre = pre.next;
  }
  pre.next = cur.next;
  cur.next = null;
  return dummy.next;
};
